### PR TITLE
feat: Use task lint for pre-commit

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments

This is so we can align projects that use this pre-commit hook with their own `task lint` command. Then, we can set vars in the user's `Taskfile.yml` and assess that automatically, while avoiding the use of `env`s which can be different in a local vs pipeline, etc.

It also simplifies the use of `goat` configurations like exemptions; you can set that in your `Taskfile.yml` instead of trying to using the `pre-commit` [`exclude`](https://pre-commit.com/#top_level-exclude) feature which can be divergent from the goat's [`exclude`](https://github.com/SeisoLLC/goat#customizations) feature, and confusing on which to pick.

## Pull Request Checklist

Thank you for submitting a contribution to the goat!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [X] Rebase your branch against the latest commit of the target branch
- [X] If you are adding a dependency, please explain how it was chosen
- [X] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [X] If there is an issue associated with your Pull Request, link the issue to the PR.

### CODE REVIEW

File: Task/python/Taskfile.yml [hunk 1]
```diff
@@ -27,8 +27,19 @@ tasks:
   lint:
     preconditions:
       - sh: docker pull seiso/goat:latest || echo "Failed to pull the latest goat, continuing anyway..."
+    # Adjustments are passed as vars, not envs, to avoid local workarounds
```
Comment (ChiefHolland) [inline Task/python/Taskfile.yml:30]: <3

File: entrypoint.sh [hunk 2]
```diff
@@ -63,6 +63,18 @@ function setup_environment() {
 
 	export REPO_DICTIONARY="${RELATIVE_PATH}/.github/etc/dictionary.txt"
 
+	#############
```
Comment (ChiefHolland) [inline entrypoint.sh:66]: Thanks for this reminder!

**Review by seanctech — DISMISSED** [review]
LGTM

**Review by seanctech — APPROVED** [review]
LGTM

**Review by ChiefHolland — APPROVED** [review]
LGTM

**Pull request merged**